.github/workflows/test-e2e.yml

```diff
@@ -16,12 +16,23 @@ jobs:
         uses: actions/setup-go@v5
         with:
           go-version-file: go.mod
+          cache: true
+
+      - name: Cache kind binary
+        uses: actions/cache@v4
+        with:
+          path: /usr/local/bin/kind
+          key: kind-${{ runner.os }}-latest
+          restore-keys: |
+            kind-${{ runner.os }}-
 
       - name: Install the latest version of kind
         run: |
-          curl -Lo ./kind https://kind.sigs.k8s.io/dl/latest/kind-linux-amd64
-          chmod +x ./kind
-          sudo mv ./kind /usr/local/bin/kind
```

Comment:
Add support for validating the cached version is the latest version, if not download the latest version and update the cache